roles/ceph-validate/tasks/check_devices.yml

```diff
@@ -29,10 +30,11 @@
   when:
     - osd_scenario == 'non-collocated'
     - item.failed
+    - dedicated_devices is defined
```

Comment:
why? `dedicated_devices` is always defined as an empty list, see: https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-osd/defaults/main.yml#L133